src/tree/SubscriptionTreeItem.ts

```diff
@@ -76,6 +81,29 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         return result.data;
     }
 
+    public async pickResourceGroup(context: IActionContext, options: PickResourceGroupOptions & { canPickMany: true }): Promise<ResourceGroupTreeItem[]>;
+    public async pickResourceGroup(context: IActionContext, options: PickResourceGroupOptions & { canPickMany: false }): Promise<ResourceGroupTreeItem>;
+    public async pickResourceGroup(context: IActionContext, options: PickResourceGroupOptions): Promise<ResourceGroupTreeItem | ResourceGroupTreeItem[]> {
+        if (this.cache.resourceGroups.length === 0) {
+            const client: ResourceManagementClient = await createResourceClient([context, this.subscription]);
+            this.cache.resourceGroups = await uiUtils.listAllIterator(client.resourceGroups.list());
+        }
+
+        const tis = (await context.ui.showQuickPick(this.cache.resourceGroups.sort((a, b) => a.name.localeCompare(b.name)).map((rg: ResourceGroupTreeItem): IAzureQuickPickItem<ResourceGroupTreeItem> => ({
```

Comment:
Just to double check, if there are zero resource groups in the subscription, this will throw the `NoResourcesFoundError`, correct? (And thus, not go on to the logic below)

Reply:
It does now!